bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ServerHandler.java

```diff
@@ -100,9 +112,9 @@ void initializeProperties(Map<String, String> properties) throws FreeboxExceptio
                         .withType(new ChannelTypeUID(BINDING_ID + ":fanspeed")).build());
             }
         });
-        if (nbInit != channels.size()) {
-            updateThing(editThing().withChannels(channels).build());
-        }
+
+        // And finally update the thing with appropriate channels
+        updateThing(editThing().withChannels(channels).build());
```

Comment:
Except if the core framework properly handles no change of channels, this is better to check if at least one channel was added or removed.
Core framework to be checked.

Reply:
I did this previously because only dynamic (sensors, fans...) channels were added.
Now for sure channels will be removed depending upon the type of media used (FTTH, xDSL).
So you can be sure that at this point channels will have changed.